### PR TITLE
crud functionality added.

### PR DESCRIPTION
**Pull request by DilsherB**
in this pull request I have added following functionality:
1. adding new item.
2. delete on click at trash icon.
3. on check the `check box` text with be strike through.
4. update the item on hitting enter.
5. delete all checked items on clicking `clear all completed`.
6. at every step `local storage` is updated.
7. updating IDs on deleting one or more items/tasks.

### CODE REVIEW

**Comment by thecodechaser** [comment]
@DilsherB please check slack for my message

File: src/index.js [hunk 1]
```diff
@@ -44,4 +42,26 @@ for (let i = 0; i < deleteIcon.length; i += 1) {
...
+    }
+    localStorage.setItem('newItem', JSON.stringify(tasks));
+  });
+  currentTask[i].addEventListener('keydown', (e) => {
+    if (e.code === 'Enter') {
+      tasks[i].description = currentTask[i].innerHTML;
+      localStorage.setItem('newItem', JSON.stringify(tasks));
+      window.location.reload();
+    }
+  });
```
Comment (thecodechaser) [inline src/index.js]: - [ ] Please make sure deleting a task should update all remaining items' indexes/ids, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1) as requested in the project requirements. Currently deleting a task doesn't update the unique indexes/ids of the tasks.
  - As in the below image you can see I deleted the first task from the list, and the index/ids of the remaining tasks are not updated and this is causing the duplication of id after adding a new task. Please make sure to fix this issue.
image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @DilsherB👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (DilsherB) [reply src/index.js]: @thecodechaser thanks for your kind review.
actually i spent much time to develop this functionality but couldn't. If you can please help me in it.

Comment (thecodechaser) [reply src/index.js]: @DilsherB I would suggest after remove event
Start a loop for the list array and update the ids of all items in the list with i+1

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @DilsherB 👋, 

Well done for implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**